src/conv/k-g/region_list.cpp

```diff
@@ -96,23 +96,32 @@ void RegionList::create
     wmember all_head;
     BU_LIST_INIT(&all_head.l);
 
+    int region_id = 0;
+
     for (std::map<std::string, Region>::iterator it = m_list.begin(); it != m_list.end(); ++it) {
 	const std::string&                        region_name              = it->first;
 	Region&                                   region                   = it->second;
 	Geometry&                                 geometry                 = region.geometry;
 	const std::map<std::string, std::string>& region_attributes        = region.attributes;
 
-	
 	std::vector<std::string> names = geometry.write(wdbp);
 
 	wmember geometry_head;
 	BU_LIST_INIT(&geometry_head.l);
 
+	bu_color regionColor = BU_COLOR_INIT_ZERO;
+	bu_color_rand(&regionColor, BU_COLOR_RANDOM);
+
+	unsigned char rgb[3];
+	bu_color_to_rgb_chars(&regionColor, rgb);
+
 	for (size_t i = 0; i < names.size(); i++) {
 	    mk_addmember(names[i].c_str(), &(geometry_head.l), NULL, WMOP_UNION);
 	    mk_addmember(names[i].c_str(), &(all_head.l), NULL, WMOP_UNION);
 	}
-	mk_lfcomb(wdbp, geometry.getBaseName(), &geometry_head, 0);
+
+	mk_lcomb(wdbp, geometry.getBaseName(), &geometry_head, 1, NULL, NULL, rgb, region_id);
+	++region_id;

```

Comment:
One more (new) thing :smiling_imp:: Please, start the region IDs with 1. You can do this by writing ++region_id inside the mk_lcomb: `mk_lcomb(wdbp, geometry.getBaseName(), &geometry_head, 1, NULL, NULL, rgb, ++region_id);`

Reply:
sure, will update.